AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/functionsFlow.js

```diff
@@ -543,13 +557,153 @@ async function validateConnectionStringAsync(propertyName, connectionString, typ
+                break;
+            case "ServiceUriMissed":
+                title = `The ${propertyName} ServiceUri setting  was not found or is set to a blank value <a href= "https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#common-properties-for-identity-based-connections" target="_blank">Click here to know more</a>`;
+                break;
+            case "SystemAssignedManagedIdentity":
+                title = "Authentication failure";
+                detailsMarkdown = `The target service is not provided with Access to the function app using system assigned identity. <a href= "https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#configure-an-identity-based-connection" target="_blank">Click here to know more</a>`;
+                break;
+            case "UserAssignedManagedIdentity":
+                title = "Authentication failure";
+                detailsMarkdown = `The target service is not provided with access to the user assigned identity which is configured for the function app. <a href= "https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#configure-an-identity-based-connection" target="_blank">Click here to know more</a>`;
+                break;
+            case "ManagedIdentityCredential":
+                title = 'Your app is not having managed identity configured for making a successful connection. <a href= "https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#configure-an-identity-based-connection" target="_blank">Click here to know more</a>';
+                break;
+            case "Forbidden":
+                // Some authentication failures come through as Forbidden so check the exception data
+                if(checkConnectionStringResult.Exception != undefined && 
+                   checkConnectionStringResult.Exception.RequestInformation != undefined && 
+                   JSON.stringify(checkConnectionStringResult.Exception.RequestInformation).includes("AuthenticationFailed")) {
+                    title = "Authentication failure";
+                    detailsMarkdown = `Authentication failure - the credentials in the configured connection string are either invalid or expired. Please update the app setting with a valid connection string.`;
+                } else {
+                    title = `Access to the ${service} resource is restricted.`;
+                    detailsMarkdown = title + `\r\n\r\n` + `This can be due to firewall rules on the resource.  Please check if you have configured firewall rules or a private endpoint and that they correctly allow access from the Function App.  Relevant documentation:`
+                        + `\r\n\r\n`;
+                    switch (type) {
+                        case ConnectionStringType.StorageAccount:

```

Comment:
Unnecessary duplicate code. The detailsMarkdown is the same for StorageAccount, BlobStorageAccount, QueueStorageAccount and FileShareStorageAccount. Use the switch case syntax to have the same logic for all four types.

Reply:
fixed